src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -233,6 +233,43 @@ public ISqlConnectionFactory ConnectionFactory
         /// <param name="connectionParams">The params to validate</param>
         /// <returns>A ConnectionCompleteParams object upon validation error, 
         /// null upon validation success</returns>
+
+        internal async Task<bool> TryRefreshAuthToken(string ownerUri)
+        {
+            if (!this.TryFindConnection(ownerUri, out ConnectionInfo connection))
+            {
+                return false;
+            }
+
+            var requestMessage = new RefreshTokenNotificationParams
+            {
+                AccountId = connection.ConnectionDetails.GetOptionValue("azureAccount", string.Empty),
+                Authority = connection.ConnectionDetails.GetOptionValue("azureTenantId", string.Empty),
+                Provider = "Azure",
+                Resource = "SQL",
+                Uri = ownerUri
+            };
+            try {
+                await this.ServiceHost.SendEvent(RefreshTokenNotification.Type, requestMessage);
+            }
+            catch (Exception e){
+                Console.WriteLine(e);

```

Comment:
Do not use Console.WriteLine here for 2 reasons:
1. the STDIO are used for IPC communication in STS, writing unexpected stuff to it will break STS
2. writing to console is effectively the same as do nothing, we won't know what happened.
Please use logger here, you can ask @Charles-Gagnon for how to use the logger.

Reply:
And add context to the message you write indicating what specifically failed